app/uk/gov/hmrc/transitmovementsvalidator/services/ValidationService.scala

```diff
@@ -109,15 +93,36 @@ class ValidationServiceImpl @Inject() (implicit ec: ExecutionContext) extends Va
               }
               .leftMap {
                 exc =>
-                  NonEmptyList.of(SchemaValidationError.fromSaxParseException(exc))
+                  NonEmptyList.of(XmlSchemaValidationError.fromSaxParseException(exc))
               }
 
             NonEmptyList
               .fromList(errorBuffer.toList)
               .map(Either.left)
               .getOrElse(parseXml)
         }
+    }
 
+  override def validateJSON(messageType: String, source: Source[ByteString, _])(implicit
+    materializer: Materializer,
+    ec: ExecutionContext
+  ): Future[Either[NonEmptyList[ValidationError], Unit]] =
+    MessageTypeJson.values.find(_.code == messageType) match {
```

Comment:
There's quite a bit of common code between the Json and XML up until 
case Some(mType) = 

if it's not too difficult, I would possibly refactor.